src/js/actions/layers.js

```diff
@@ -1694,10 +1684,13 @@ define(function (require, exports) {
 
         var numberRef = (typeof target === "number"),
             targetRef = numberRef ? layerLib.referenceBy.index(target) : layerLib.referenceBy[target],
-            reorderObj = layerLib.reorder(layerRef, targetRef),
-            reorderPromise = descriptor.playObject(reorderObj);
+            reorderObj = layerLib.reorder(layerRef, targetRef);
       
-        return reorderPromise.bind(this)
+        return this.transfer(historyActions.newHistoryState, document.id, strings.ACTIONS.REORDER_LAYERS)
```

Comment:
Just to confirm: it's important that this run before we play the action below? Is there any opportunity for more concurrency?